modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/StandardHeader.scala

```diff
@@ -208,43 +209,64 @@ class StandardHeader(
   hs: DhsClient,
   obsReader: ObsKeywordsReader,
   tcsReader: TcsKeywordsReader,
+  gwsReader: GwsKeywordReader,
   stateReader: StateKeywordsReader) extends Header {
 
   import Header._
+  import Header.Defaults._
+  import KeywordsReader._
 
   override def sendBefore(id: ImageFileId, inst: String): SeqAction[Unit] = {
 
-    val p: SeqAction[Double] = for {
-      xoff <- tcsReader.getXOffset
-      yoff <- tcsReader.getYOffset
-      iaa  <- tcsReader.getInstrumentAA
+    val p: SeqAction[Option[Double]] = for {
```

Comment:
There seems to be some repetition on the calculations of `p`, `q`, etc. Are there any opportunities to reduce duplicated code?